src/main/scala/sigmastate/lang/SigmaTyper.scala

```diff
@@ -208,7 +209,7 @@ class SigmaTyper {
         case nl: SNumericType => (m, newArgs) match {
           case ("*", Seq(r)) => r.tpe match {
             case nr: SNumericType =>
-              bimap(env, "*", newObj.asNumValue, r.asNumValue)(Multiply)(tT, tT)
+              bimap(env, "*", newObj.asNumValue, r.asNumValue)(TransformingSigmaBuilder.Multiply)(tT, tT)
```

Comment:
@greenhat Same here, let's pass a builder to parser, binder, typer, etc. everywhere we need to construct new IR nodes